ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -196,10 +202,12 @@ private Callback<Void> securityPostProcessRequestCallback(BlobInfo blobInfo) {
               new RetainingAsyncWritableChannel(frontendConfig.maxJsonRequestSizeBytes);
           restRequest.readInto(channel, fetchStitchRequestBodyCallback(channel, blobInfo));
         } else {
+          if (RestUtils.isDatasetUpload(restRequest.getArgs())) {
+            addDatasetVersion(blobInfo.getBlobProperties(), restRequest);
+          }
```

Comment:
When do we add dataset version to data?
We have three operations in this handler.
1. Call router.putBlob to upload blob to ambry-servers
2. Add blobName and blob id mapping in the named blob db 
3. Add dataset version in the account db

I think we should follow this order to make sure when dataset version is created in the account db, the named blob name to id mapping is already created in named blob db.

Reply:
As we discussed offline, I'm planning to use Dataset DB to be SOT in order to support the latest version feature. Otherwise it need long turn around time for people to know what's the current latest version. We might have some orphan record in dataset DB if put failed, and in future rb, we will try to delete the orphan record. If still failed, customer should leverage best effort to manually delete it. I think in this way we could prevent so many orphan record generate in DB, besides, the chance to generate orphan record is pretty low since we will try to delete first.